integration/src/integration/java/pl/allegro/tech/hermes/integration/management/OfflineRetransmissionManagementTest.java

```diff
@@ -0,0 +1,195 @@
+
+    @BeforeClass
+    public void setupClass() {
+        operations.createGroup(GROUP);
+    }
+
+    @AfterMethod
+    public void cleanup() {
+        deleteTasks();
+    }
+
+    @Test
+    public void shouldCreateRetransmissionTask() {
+        // given
+        Topic sourceTopic = createTopic();
+        Topic targetTopic = createTopic();
+
+        operations.buildTopic(sourceTopic);
+        operations.buildTopic(targetTopic);
+
+        // when
+        OfflineRetransmissionRequest request = createRequest(
+                sourceTopic.getQualifiedName(),
+                targetTopic.getQualifiedName());
+        Response response = management.offlineRetransmission().createRetransmissionTask(request);
+
+        // then
+        assertThat(response).hasStatus(Response.Status.CREATED);
+
+        // and
+        List<OfflineRetransmissionTask> allTasks = management.offlineRetransmission().getAllTasks();
+        assertThat(allTasks.size()).isEqualTo(1);
+        assertThat(allTasks.get(0).getStartTimestamp()).isEqualTo(request.getStartTimestamp());
+        assertThat(allTasks.get(0).getEndTimestamp()).isEqualTo(request.getEndTimestamp());
+        assertThat(allTasks.get(0).getSourceTopic()).isEqualTo(request.getSourceTopic());
+        assertThat(allTasks.get(0).getTargetTopic()).isEqualTo(request.getTargetTopic());
+    }
+
+
+    @Test
+    public void shouldReturnEmptyListIfThereAreNoTasks() {
+        // expect
+        assertThat(management.offlineRetransmission().getAllTasks().size()).isEqualTo(0);
+    }
+
+    @Test
+    public void shouldReturnClientErrorWhenRequestingRetransmissionWithNotExistingSourceTopic() {
+        // given
+        Topic targetTopic = createTopic();
+
+        // when
+        OfflineRetransmissionRequest request = createRequest("not.existing.topic",
+                targetTopic.getQualifiedName());
+        Response response = management.offlineRetransmission().createRetransmissionTask(request);
+
+        // then
+        assertThat(response).hasStatus(Response.Status.BAD_REQUEST);
+        assertThat(response).containsMessage("Source topic does not exist");
+    }
+
+    @Test
+    public void shouldReturnClientErrorWhenRequestingRetransmissionWithNotExistingTargetTopic() {
+        // given
+        Topic sourceTopic = createTopic();
+        operations.buildTopic(sourceTopic);
+
+        // when
+        OfflineRetransmissionRequest request = createRequest(
+                sourceTopic.getQualifiedName(), "not.existing.topic");
+        Response response = management.offlineRetransmission().createRetransmissionTask(request);
+
+        // then
+        assertThat(response).hasStatus(Response.Status.BAD_REQUEST);
+        assertThat(response).containsMessage("Target topic does not exist");
+    }
+
+    @Test
+    public void shouldReturnClientErrorWhenRequestingRetransmissionWithNegativeTimeRange() {
+        // given
+        Topic sourceTopic = createTopic();
```

Comment:
Można by było zdefiniować prywatną klasę, która zamykałaby operacje wykonywane w testach tworzących 2 topiki. Mogłoby to wyglądać tak:
~~~~
// given
OfflineRetranssmissionTestDataBuilder builder = new OfflineRetransmissionTestDataBuilder();
builder.buildTopics();

// when
OfflineRetransmissionRequest request = builder.buildRequest();
Response response = ...
...
~~~~

Mogłoby to trochę skrócić kod niektórych testów.